packages/frontend/src/stories/Tree.tsx

```diff
@@ -0,0 +1,158 @@
+    uuid: string;
+    name: string;
+    path: string;
+};
+
+const convertNestableListToTree = (items: NestableItem[]): TreeNodeData[] => {
+    const buildTree = (
+        nodes: NestableItem[],
+        parentPath = '',
+    ): TreeNodeData[] => {
+        return nodes
+            .filter((item) => {
+                const pathParts = item.path.split('.');
+                const itemParentPath = pathParts.slice(0, -1).join('.');
+                return itemParentPath === parentPath;
+            })
+            .map((item) => {
+                const children = buildTree(nodes, item.path);
+                return {
+                    label: item.name,
+                    value: item.uuid,
+                    nodeProps: { uuid: item.uuid },
+                    ...(children.length > 0 ? { children } : {}),
+                };
+            });
+    };
+
+    return buildTree(items);
+};
+
+const renderTreeNode = ({
+    node,
+    selected,
+    expanded,
+    hasChildren,
+    elementProps,
+    tree,
+}: RenderTreeNodePayload) => {
+    return (
+        <div {...elementProps}>
+            <Paper
+                component={Group}
+                miw="200px"
+                maw="300px"
+                gap={5}
+                wrap="nowrap"
+                h="32"
+                px="sm"
+                radius="md"
+                bg={selected ? 'blue.0' : 'transparent'}
+                style={{ overflow: 'hidden' }}
+                onClick={() => tree.toggleSelected(node.value)}
+            >
+                {hasChildren && (
+                    <ActionIcon
+                        className={classes.actionIcon}
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            tree.toggleExpanded(node.value);
+                        }}
+                        size="xs"
+                        variant="transparent"
+                    >
+                        <MantineIcon
+                            icon={expanded ? IconChevronDown : IconChevronRight}
+                            size="lg"
+                        />
+                    </ActionIcon>
+                )}
+
+                <MantineIcon
+                    icon={IconFolder}
+                    color="gray.7"
+                    size="lg"
+                    stroke={1.5}
+                    style={{ flexShrink: 0 }}
+                />
+
+                <Text inline truncate="end" style={{ flexGrow: 1 }}>
+                    {node.label}
+                </Text>
+
+                {selected && (
+                    <MantineIcon
+                        icon={IconCheck}
+                        size="lg"
+                        color="blue.6"
+                        style={{ flexShrink: 0 }}
+                    />
+                )}
+            </Paper>
+        </div>
+    );
+};
+
+type Props = {
+    data: NestableItem[];
+    onSelect: (selectedUuid: string | null) => void;
+};
```

Comment:
I think we want the tree to be controlled (`value/selected` prop) or at least have initial value (`initialValue/initialSelected` prop) (depending on how good `Tree/useTree` api is)

For example when moving chart/dash/space to a different space, when I click button an the tree comes up I want it to have my current location selected and scrolled to

Reply:
ok I'll add that in the next iteration!